### PR TITLE
Anthony/conditional policies

### PR DESCRIPTION
**Pull request by notanthony**

### CODE REVIEW

**Review by mstanbCO — COMMENTED** [review]
Left a couple of questions, but generally it looks good. Happy to +1 once we clear up those questions.

File: internal/provider/policy_resource_sdk.go [hunk 1]
```diff
@@ -43,17 +49,6 @@ func (r *PolicyResourceModel) ToCreateSDKType() *shared.CreatePolicyRequest {
 						AllowSelfApproval: step.Approval.AppOwnerApproval.AllowSelfApproval.ValueBoolPointer(),
 					}
 				}
-				if step.Approval.AppGroupApproval != nil {
```
[the remark this answers is not in the document]
Comment (notanthony) [reply internal/provider/policy_resource_sdk.go:46]: I got confused, thought it wasnt used, going to add it back

[the remark this answers is not in the document]
Comment (notanthony) [reply internal/provider/policy_resource_sdk.go:46]: Looked deeper, seems like its read only so its been removed

File: internal/provider/type_policy_steps.go
[the remark this answers is not in the document]
Comment (notanthony) [reply internal/provider/type_policy_steps.go:1]: It just changed the chmod, this file isn't empty, Speakeasy somethings does this its weird

**Review by mstanbCO — APPROVED** [review]

**Pull request merged**